### PR TITLE
Improving Trader Options

### PR DESCRIPTION
**Pull request by Kurfursten**
untitled

Mecha graveyard has several mechas you can salvage for parts, but the 2 RIPLEYs and CLARK have been extra beefed up to be sure to drop plenty of goodies. **Important note: the trader shuttle cannot go to the mecha graveyard, only the mining or research shuttle will fit the dock.**

Wonderful Wardrobe can end up getting stuff like thermals, vampire cape, security plasmaman suit, cat morphing mask, medal of captaincy, advanced hardsuit helmet, security sunglasses... or even just 19 random worthless jumpsuits and a maid outfit... It uses the completely unaltered list from Spessmart. This might need more blacklisting or less items, I'll let github help me decide on that. Just remember it is a premium item so it should have a chance for *something* exciting. I think 20 is a safe number because in a set I usually see 2-3 "exciting" items, like the ones mentioned above (that's everything of note from 3 wardrobe spawns)

🆑 
* tweak: Replaced box of marauder parts at the trader vendor with a destination disk for a mecha graveyard full of wonderful salvageable robotics stuff. Bring your toolbelt and a suit!
* tweak: Replaced bottle of nanobots, bottle of rezadone, and bottle of peridazon with a box containing all three
* rscadd: Added the Wonderful Wardrobe. It contains 20 random Spessmart approved clothing items.

### CODE REVIEW

**Comment by Probe1** [comment]
Balance added because thermals.

**Comment by Kurfursten** [comment]
If people want to discuss this more I can go generate more random closets to give you examples of an average haul.

The chances of getting any specific good thing (e.g.: thermals) are very low but the chances of getting at least one cool/good thing are pretty high.

**Comment by Probe1** [comment]
I personally don't have an issue with it or need statistical testing.  Adding traitor items to something still qualifies for a balance label though.

**Comment by Dacendeth** [comment]
I look foward to a trader opening a wonderful wardrobe and a live face hugger popping out.

**Comment by Kurfursten** [comment]
That is definitely a possibility. And yet... I feel like it's an acceptable risk for the same reason as stuff like thermals appearing. Buyer beware.

**Comment by Dacendeth** [comment]
Do you think it would be better if instead of those purple tiles you had the hyperspace tiles that make you get shoved out into space somewhere? Just incase you get trapped there by someone calling the shuttle away?

**Comment by Kurfursten** [comment]
Oh yeah, we could do that. Seems like a good idea.

**Comment by Kurfursten** [comment]
What's new?
* Now bordered by transit tiles that kick you out into space if you touch them
* Added an external airlock by where the mining/research shuttle docks so that it doesn't make you depressurize your ship if you left the door open (most people do)

This is ready for merge

**Review by MadmanMartian — CHANGES_REQUESTED** [review]
transit tiles alone aren't what throw you around and teleport you. Surround the map with /obj/effect/step_trigger/teleporter/random/shuttle_transit to teleport them into deep space, and /obj/effect/step_trigger/thrower for what throws you.

Make the surrounding transit turf border 7 tiles or more around, lest players break their immersions by seeing other dungeons.

File: code/game/objects/items/trader.dm [hunk 1]
```diff
@@ -22,6 +22,18 @@
...
 
+/obj/item/weapon/storage/trader_chemistry
+	name = "chemist's pallet"
+	desc = "Everything you need to make art."
+	icon = 'icons/obj/storage/smallboxes.dmi'
+	icon_state = "box_of_doom"
+	item_state = "box_of_doom"
+
+/obj/item/weapon/storage/trader_chemistry/New()
+	new /obj/item/weapon/reagent_containers/glass/bottle/peridaxon(src)
```
Comment (MadmanMartian) [inline code/game/objects/items/trader.dm]: Call ..()

File: code/game/objects/items/trader.dm [hunk 2]
```diff
@@ -35,6 +47,23 @@
...
+	req_access = list(access_trade)
+	icon_state = "cabinetdetective_locked"
+	icon_closed = "cabinetdetective"
+	icon_locked = "cabinetdetective_locked"
+	icon_opened = "cabinetdetective_open"
+	icon_broken = "cabinetdetective_broken"
+	icon_off = "cabinetdetective_broken"
+
+/obj/structure/closet/secure_closet/wonderful/New()
+	var/random_clothes = clothing.Copy()
```
Comment (MadmanMartian) [inline code/game/objects/items/trader.dm]: Call ..()

**Review by MadmanMartian — CHANGES_REQUESTED** [review]

**Comment by Kurfursten** [comment]
untitled

As you can see from the screenshots
* Hyperspace radius increased
* Hyperspace properly throws you into deep space
* Disk not compatible with Trader ship

**Comment by MadmanMartian** [comment]
we could do with an opaque spaceturf for spacebound dungeons in the future

**Review by MadmanMartian — APPROVED** [review]

**Pull request merged**